### PR TITLE
fix custom composer vendor-dir loading

### PR DESCRIPTION
**Pull request by o0h**
## Purpose
With composer config "vendor-dir", AbstractAnnotator fail to manual autoload for php_codesniffer.
To fix this,  find composer package directory path in ancestor instead of 'getcwd() . '/vendor/' . 

## What happened
in composer.json

```json
{
    "config": {
        "vendor-dir": "vendor/php-composer"
    }
}
```

run `bin/cake  IdeHelper.annotations all` to throw exception.
```
bash-4.2# bin/cake  IdeHelper.annotations all
[Models]
Warning Error: Class '\PHP_CodeSniffer_File' not found in [/var/www/html/vendor/composer/dereuromark/cakephp-ide-helper/src/Annotator/AbstractAnnotator.php, line 29]

2017-12-14 11:34:47 Warning: Warning (2): Class '\PHP_CodeSniffer_File' not found in [/var/www/html/vendor/composer/dereuromark/cakephp-ide-helper/src/Annotator/AbstractAnnotator.php, line 29]
Trace:
Cake\Error\BaseErrorHandler::handleError() - CORE/src/Error/BaseErrorHandler.php, line 153
class_alias - [internal], line ??
include - ROOT/vendor/composer/dereuromark/cakephp-ide-helper/src/Annotator/AbstractAnnotator.php, line 29
Composer\Autoload\includeFile - ROOT/vendor/composer/composer/ClassLoader.php, line 444
Composer\Autoload\ClassLoader::loadClass() - ROOT/vendor/composer/composer/ClassLoader.php, line 322
```

### What I did
As `vendor-dir`path, set two level upper from plugin directory.

### CODE REVIEW

File: src/Annotator/AbstractAnnotator.php [hunk 1]
```diff
@@ -21,7 +21,10 @@
 use RuntimeException;
 use SebastianBergmann\Diff\Differ;
 
-$manualAutoload = getcwd() . '/vendor/squizlabs/php_codesniffer/autoload.php';
+$currentDir = __DIR__;
+$vendorDir = substr($currentDir, 0, strpos($currentDir, '/cakephp-ide-helper'));
```
Comment (dereuromark) [inline src/Annotator/AbstractAnnotator.php]: Could this fail on windows? maybe use DS here when you compare strings.

**Comment by dereuromark** [comment]
Can you please check if the change works for you? Hopefully this makes the tests pass again.

**Comment by o0h** [comment]
case:  `vendor/` directory exists and  php composer vendor-dir is set to `vendor/php-composer` , `$manualAutoload` will be invalid path 😢 

* expect `vendor/php-composer/squizlabs/php_sniffer/autoload.php`
* actual `vendor/squizlabs/php_codesniffer/autoload.php`

(my team also use ruby bundler, so there are `vendor/bundle` and `vendor/php-composer`.

**Comment by dereuromark** [comment]
Can u reverse the lookup order?

**Comment by o0h** [comment]
i tried to fix https://github.com/dereuromark/cakephp-ide-helper/pull/66#issuecomment-351688644

**Pull request merged**